plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -459,7 +459,15 @@ private Document getTableMetadata(String schemaName, String tableName)
                 Document metadata = new Document(TABLE_NAME_KEY, tableName);
                 metadata.append(FIELDS_KEY, guessTableFields(schemaName, tableName));
 
-                schema.createIndex(new Document(TABLE_NAME_KEY, 1), new IndexOptions().unique(true));
+                boolean isSchemaCollectionIndexed = getIndexes(schemaName, tableName).stream()
+                        .anyMatch(mongoIndex -> mongoIndex.getKeys().stream()
+                                .anyMatch(mongodbIndexKey -> TABLE_NAME_KEY.equals(mongodbIndexKey.getName()))
+                        );
+
+                if (!isSchemaCollectionIndexed) {
+                    schema.createIndex(new Document(TABLE_NAME_KEY, 1), new IndexOptions().unique(true));
```

Comment:
There is another usage of `createIndex()`. Please extract a method and apply to there. 
```java
    private boolean indexExists(MongoCollection<Document> schemaCollection) {
        for (MongoIndex index : MongoIndex.parse(schemaCollection.listIndexes())) {
            List<MongoIndex.MongodbIndexKey> keys = index.getKeys();
            if (keys.size() == 1 && TABLE_NAME_KEY.equals(keys.get(0).getName())) {
                return true;
            }
        }
        return false;
    }
```